drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,663 @@
+{
+    gpio_set(dev->params.pin_ce);
+    xtimer_usleep(NRF24L01P_NG_DELAY_US_CE_HIGH_PULSE);
+    gpio_clear(dev->params.pin_ce);
+    xtimer_usleep(NRF24L01P_NG_DELAY_US_TX_SETTLING);
+}
+
+static
+int _assert_awake(const nrf24l01p_ng_t *dev)
+{
+    uint8_t config = 0;
+    nrf24l01p_ng_read_reg(dev, NRF24L01P_NG_REG_CONFIG, &config, 1);
+    return config & NRF24L01P_NG_FLG_PWR_UP;
+}
+
+static
+netopt_state_t _state_to_netif(nrf24l01p_ng_state_t state)
+{
+    if (state == NRF24L01P_NG_STATE_POWER_DOWN) {
+        return NETOPT_STATE_SLEEP;
+    }
+    if (state == NRF24L01P_NG_STATE_STANDBY_1) {
+        return NETOPT_STATE_STANDBY;
+    }
+    if (state == NRF24L01P_NG_STATE_STANDBY_2) {
+        return NETOPT_STATE_TX;
+    }
+    if (state == NRF24L01P_NG_STATE_TX_MODE) {
+        return NETOPT_STATE_TX;
+    }
+    if (state == NRF24L01P_NG_STATE_RX_MODE) {
+        return NETOPT_STATE_RX;
+    }
+    return NETOPT_STATE_OFF; /* error */
+}
+
+nrf24l01p_ng_state_t _state_from_netif(netopt_state_t state)
+{
+    if (state == NETOPT_STATE_SLEEP) {
+        return NRF24L01P_NG_STATE_POWER_DOWN;
+    }
+    if (state == NETOPT_STATE_STANDBY) {
+        return NRF24L01P_NG_STATE_STANDBY_1;
+    }
+    if (state == NETOPT_STATE_TX) {
+        return NRF24L01P_NG_STATE_TX_MODE;
+    }
+    if (state == NETOPT_STATE_RX) {
+        return NRF24L01P_NG_STATE_RX_MODE;
+    }
+    return NRF24L01P_NG_STATE_UNDEFINED;
+}
+
+static
+void nrf24l01p_ng_irq_handler(void *_dev)
+{
+    nrf24l01p_ng_t *dev = (nrf24l01p_ng_t *)_dev;
+    /* Once the IRQ pin has triggered,
+       do not congest the thread´s
+       message queue with IRQ events */
+    gpio_irq_disable(dev->params.pin_irq);
+    netdev_trigger_event_isr((netdev_t *)dev);
+}
+
+static
+void _isr_max_rt(nrf24l01p_ng_t *dev)
+{
+    assert(dev->state == NRF24L01P_NG_STATE_STANDBY_1 ||
+           dev->state == NRF24L01P_NG_STATE_STANDBY_2 ||
+           dev->state == NRF24L01P_NG_STATE_RX_MODE   ||
+           dev->state == NRF24L01P_NG_STATE_TX_MODE);
+    DEBUG_PUTS("[nrf24l01p_ng] IRS MAX_RT");
+    nrf24l01p_ng_flush_tx(dev);
+    dev->netdev.event_callback(&dev->netdev, NETDEV_EVENT_TX_NOACK);
+}
+
+static
+void _isr_rx_dr(nrf24l01p_ng_t *dev)
+{
+    assert(dev->state == NRF24L01P_NG_STATE_STANDBY_1 ||
+           dev->state == NRF24L01P_NG_STATE_STANDBY_2 ||
+           dev->state == NRF24L01P_NG_STATE_RX_MODE   ||
+           dev->state == NRF24L01P_NG_STATE_TX_MODE);
+    DEBUG_PUTS("[nrf24l01p_ng] IRS RX_DR");
+    uint8_t fifo_status;
+    nrf24l01p_ng_read_reg(dev, NRF24L01P_NG_REG_FIFO_STATUS, &fifo_status, 1);
+    /* read all RX data */
+    while (!(fifo_status & NRF24L01P_NG_FLG_RX_EMPTY)) {
+        DEBUG_PUTS("[nrf24l01p_ng] ISR: read pending Rx frames");
+        dev->netdev.event_callback(&dev->netdev, NETDEV_EVENT_RX_COMPLETE);
+        nrf24l01p_ng_read_reg(dev, NRF24L01P_NG_REG_FIFO_STATUS,
+                              &fifo_status, 1);
+    }
```

Comment:
How does that work?
I assume the status register is cleared by reading it, but where will the framebuffer be read?

Reply:
`dev->netdev.event_callback(&dev->netdev, NETDEV_EVENT_RX_COMPLETE);`
This causes my `_recv()` function to be called:
```C
        switch (event) {
            case NETDEV_EVENT_RX_COMPLETE:
                pkt = netif->ops->recv(netif);
                if (pkt) {
                    _pass_on_packet(pkt);
                }
                break;
```

The interrupts in the status regiser are cleared, if you write `1` to it.
I do that after I handled the interrupts in `_isr()`:
```C
    nrf24l01p_ng_write_reg(dev, NRF24L01P_NG_REG_STATUS, &status, 1);
```

Here is what the datasheet reads:
> The RX_DR IRQ is asserted by a new packet arrival event. The procedure for handling this interrupt should be: 1) read payload through SPI, 2) clear RX_DR IRQ, 3) read FIFO_STATUS to check if there are more payloads available in RX FIFO, 4) if there are more data in RX FIFO, repeat from step 1)